R/utils.R

```diff
@@ -0,0 +1,60 @@
+# Getting file extension
+get_file_ext <- function(file_paths) {
+  sapply(file_paths, function(file_path) {
+    file_name <- basename(file_path)
+    file_parts <- strsplit(file_name, "\\.")[[1]]
+    file_extension <- tail(file_parts, n = 1)
+    return(file_extension)
+  })
+}
+
+# Function to scale a numeric vector to percentage
+scale_to_percent <- function(x, digits = 2) {
+  percent_values <- x * 100
+  formatted_percent_values <- sprintf(paste0("%.", digits, "f%%"), percent_values)
+  return(formatted_percent_values)
+}
+
+# Function to replace NA values with the last non-NA value
+replace_na_with_last <- function(x) {
+  last_non_na <- NA
+  for (i in seq_along(x)) {
+    if (!is.na(x[i])) {
+      last_non_na <- x[i]
+    } else if (!is.na(last_non_na)) {
+      x[i] <- last_non_na
+    }
+  }
+  return(x)
+}
+
+# Function to get relative path
+path_rel <- function(path, start = ".") {
```

Comment:
I had a problem when path and start is the same, or start is a subdirectory of path.
```r
path_rel("a/b", "a")
#> [1] "b"
path_rel("a", "a")
#> [1] "NA/a"
path_rel("a", "a/b")
#> [1] "../NA/a"
```

Reply:
Nice solution :)